web/src/lib.rs

```diff
@@ -782,18 +789,36 @@ impl RuffleHandle {
                     INSTANCES.with(|instances| {
                         if instances.borrow().len() >= 8 {
                             let _ = ruffle.remove_instance();
-                            js_player.reload_with_canvas_renderer();
+                            js_player_callback.reload_with_canvas_renderer();
                         }
                     });
                 },
             ));
 
-            instance.unload_callback =
-                Some(JsCallback::register(&window, "unload", false, move |_| {
+            // Create pageshow event handler.
+            let js_player_callback = js_player.clone();
+            instance.pageshow_callback = Some(JsCallback::register(
+                &window,
+                "pageshow",
+                false,
+                move |js_event: PageTransitionEvent| {
+                    if js_event.persisted() {
+                        js_player_callback.display_restored_from_bfcache_message();
+                    }
+                },
+            ));
+
+            // Create pagehide event handler.
+            instance.pagehide_callback = Some(JsCallback::register(
+                &window,
+                "pagehide",
+                false,
+                move |_js_event: PageTransitionEvent| {
                     let _ = ruffle.with_core_mut(|core| {
                         core.flush_shared_objects();
```

Comment:
I think this shouldn't be needed, as we do that when dropping the player.

Reply:
(that's not blocking obviously, but we could get rid of this callback altogether I think)

Reply:
The player is not dropped when leaving the page.

Reply:
Aw :/ it's interesting why the player isn't being dropped. Nevermind then